blockchain/difficulty.go

```diff
@@ -540,6 +540,69 @@ func (b *BlockChain) sumPurchasedTickets(startNode *blockNode, numToSum int64) i
 	return numPurchased
 }
 
+// immatureTickets returns the total number of immature tickets at the
+// current tip and a map of immature ticket counts keyed by the block height at
+// which they were bought at.
+func (b *BlockChain) immatureTickets() (int64, map[int64]int64, error) {
+	tip := b.bestChain.Tip()
+	rangeStart := tip.height - int64(b.chainParams.TicketMaturity)
+	startNode := b.bestChain.nodeByHeight(rangeStart)
```

Comment:
Since this walks the chain through `node.parent`, should we instead start iterating at the newest block in the range rather than the oldest?

Reply:
You're right, good catch.